python/vision_explanation_methods/evaluation/pointing_game.py

```diff
@@ -148,15 +172,15 @@ def calculate_gt_salient_pixel_overlap(self,
         :return: return percent of salient pixel overlap with the ground truth
         :rtype: Float
         """
-        saliency_scores = torch.tensor(saliency_scores)
+        saliency_scores = torch.tensor(saliency_scores)[0]
         gt_bbox = torch.tensor(gt_bbox)
 
         gt_mask = torch.zeros_like(saliency_scores, dtype=torch.bool)
-        gt_mask[gt_bbox[0]:gt_bbox[2], gt_bbox[1]:gt_bbox[3]] = True
+        gt_mask[gt_bbox[1]:gt_bbox[3]+1, gt_bbox[0]:gt_bbox[2]+1] = True
 
         positive_mask = torch.gt(saliency_scores, 0)
         positive_gt_mask = torch.logical_and(positive_mask, gt_mask)
-        good = positive_gt_mask.sum()
-        total = positive_mask.sum()
+        good = torch.count_nonzero(positive_gt_mask)
+        total = torch.count_nonzero(gt_mask)
 
         return torch.div(good, total).item()
```

Comment:
Side note: Assuming `.item()` doesn't need device detachment with a GPU etc. Feel free to also look up online to generally verify.

Reply:
good point - looks like since .item() changes the object type to no longer be a tensor, the change to CPU device is automatic. Reference - https://discuss.pytorch.org/t/does-item-automatically-move-the-data-to-the-cpu/69629